packages/palette_generator/lib/palette_generator.dart

```diff
@@ -157,34 +155,32 @@ class PaletteGenerator with Diagnosticable {
   /// The [imageProvider] and [timeout] arguments must not be null.
   static Future<PaletteGenerator> fromImageProvider(
     ImageProvider imageProvider, {
-    Size size,
-    Rect region,
-    int maximumColorCount,
-    List<PaletteFilter> filters,
-    List<PaletteTarget> targets,
+    Size? size,
+    Rect? region,
+    int? maximumColorCount,
+    List<PaletteFilter>? filters,
+    List<PaletteTarget>? targets,
     Duration timeout = const Duration(seconds: 15),
   }) async {
-    assert(imageProvider != null);
-    assert(timeout != null);
-    assert(region == null || (region != null && size != null));
+    assert(region == null || size != null);
```

Comment:
if `region != null` and `size == null` then this assert check fails. But I don't see mentioned this case in docs. How should it be handled?

Reply:
If there is a defined region, then there needs to be a defined size.